app/src/main/java/com/example/voyagerx/LaunchDetailsFragment.kt

```diff
@@ -0,0 +1,111 @@
+package com.example.voyagerx
+
+import android.os.Bundle
+import android.util.Log
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import coil.load
+import coil.size.Precision
+import coil.size.Scale
+import com.example.voyagerx.databinding.FragmentLaunchDetailsBinding
+import com.example.voyagerx.repository.model.Launch
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [LaunchDetailsFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class LaunchDetailsFragment : Fragment() {
+
+    private var _binding: FragmentLaunchDetailsBinding? = null
+    private val binding get() = _binding!!
+
+    //placeholder data until we get landing page connected
+    private var launchObj : Launch = Launch(
+        "1",
+        "Thaicom 6",
+        "Cape Canaveral Air Force Station Space Launch Complex 40",
+        "2020-12-13T17:30:00.000Z",
+        "2020",
+        "SpaceX will launch Sentinel-6 Michael Freilich into low Earth orbit for NASA, NOAA, ESA, and the European Organization for the Exploitation of Meteorological Satellites aboard a Falcon 9 from SLC-4E, Vandenberg Air Force Station. Sentinel-6(A) is an ocean observation satellite providing radar ocean surface altimetry data and also atmospheric temperature profiles as a secondary mission. The booster for this mission is will land at LZ-4.",
+        "https://spaceflightnow.com/2020/11/21/international-satellite-launches-to-extend-measurements-of-sea-level-rise/",
+        "https://youtu.be/aVFPzTDCihQ",
+        listOf("https://live.staticflickr.com/65535/50630802488_8cc373728e_o.jpg",
+            "https://live.staticflickr.com/65535/50631642722_3af8131c6f_o.jpg",
+            "https://live.staticflickr.com/65535/50631544171_66bd43eaa9_o.jpg",
+            "https://live.staticflickr.com/65535/50631543966_e8035d5cca_o.jpg",
+            "https://live.staticflickr.com/65535/50631643257_c214ceee7b_o.jpg",
+            "https://live.staticflickr.com/65535/50631643917_cb7db291d0_o.jpg")
+    )
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentLaunchDetailsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        displayLaunchDetails()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    private fun displayLaunchDetails() {
+        launchObj.image_links?.let {
+            var imgUrl = launchObj.image_links?.elementAt(0)
+            binding.ivLaunchPictures.load(imgUrl)
+        } ?: run {
```

Comment:
I'm still learning Kotlin and thought this was pretty interesting, how does this `?: run` block work after the `let` statement?

Reply:
The first symbol is called a "Elvis" operator. It essentially means "if the expression on the left is null, then assign then return this logic on the right side". - https://kotlinlang.org/docs/null-safety.html#elvis-operator

As for "run" - https://kotlinlang.org/docs/scope-functions.html#run

Reply:
If `launchObj.image_links` is null, the safe call `?.` will return null for the entire left side of the elvis operator (the `let` block). The elvis operator will then run the right side.

I found this SO post helpful: https://stackoverflow.com/questions/34498562/in-kotlin-what-is-the-idiomatic-way-to-deal-with-nullable-values-referencing-o

Reply:
Interesting, thank you! Reminds me of the nullish operator in javascript.